iina/VideoView.swift

```diff
@@ -383,7 +360,9 @@ extension VideoView {
     videoLayer.wantsExtendedDynamicRangeContent = true
     videoLayer.colorspace = CGColorSpace(name: name!)
     mpv.setString(MPVOption.GPURendererOptions.iccProfile, "")
-    mpv.setString(MPVOption.GPURendererOptions.targetTrc, gamma)
+    mpv.setString(MPVOption.GPURendererOptions.toneMapping, "linear")
+    mpv.setDouble(MPVOption.GPURendererOptions.toneMappingParam, peak)
```

Comment:
Are you sure what you do is correct?

Reply:
With `extendedLinear` colorspace the output should be linear value with SDR reference white point at `1.0`. 

What these parameters do is:

- `target-trc=linear` makes mpv linearize input signal with SDR reference white at `1.0` ([see video_shaders.c line 860](https://github.com/mpv-player/mpv/blob/v0.34.1/video/out/gpu/video_shaders.c#L860))
- but because mpv believes `linear` should only be SDR range 0~1.0, it will then try to tone map ([see line 872, `dst.sig_peak` would be 1.0](https://github.com/mpv-player/mpv/blob/v0.34.1/video/out/gpu/video_shaders.c#L872))
- `tone-mapping=linear` makes tone mapping a linear scaling, and `tone-mapping-param=peak` makes the value input peak signal (`> 1.0` SDR reference white) scale to itself, thereby preserving SDR reference white at `1.0` ([see line 766](https://github.com/mpv-player/mpv/blob/v0.34.1/video/out/gpu/video_shaders.c#L766))

After checking shader used by mpv to render video in `mpv.log`, it does behave like what I want.

<details><summary>output shader in mpv.log</summary>

```
[  11.683][d][libmpv_render] fragment shader source:
[  11.684][d][libmpv_render] [  1] #version 410
[  11.684][d][libmpv_render] [  2] #define tex1D texture
[  11.684][d][libmpv_render] [  3] #define tex3D texture
[  11.684][d][libmpv_render] [  4] #define LUT_POS(x, lut_size) mix(0.5 / (lut_size), 1.0 - 0.5 / (lut_size), (x))
[  11.684][d][libmpv_render] [  5] out vec4 out_color;
[  11.684][d][libmpv_render] [  6] in vec2 texcoord0;
[  11.684][d][libmpv_render] [  7] in vec2 texcoord1;
[  11.684][d][libmpv_render] [  8] uniform mat3 colormatrix;
[  11.684][d][libmpv_render] [  9] uniform vec3 colormatrix_c;
[  11.684][d][libmpv_render] [ 10] uniform vec3 src_luma;
[  11.684][d][libmpv_render] [ 11] uniform vec3 dst_luma;
[  11.684][d][libmpv_render] [ 12] uniform sampler2DRect texture0;
[  11.684][d][libmpv_render] [ 13] uniform vec2 texture_size0;
[  11.684][d][libmpv_render] [ 14] uniform mat2 texture_rot0;
[  11.684][d][libmpv_render] [ 15] uniform vec2 texture_off0;
[  11.684][d][libmpv_render] [ 16] uniform vec2 pixel_size0;
[  11.684][d][libmpv_render] [ 17] uniform sampler2DRect texture1;
[  11.684][d][libmpv_render] [ 18] uniform vec2 texture_size1;
[  11.684][d][libmpv_render] [ 19] uniform mat2 texture_rot1;
[  11.684][d][libmpv_render] [ 20] uniform vec2 texture_off1;
[  11.684][d][libmpv_render] [ 21] uniform vec2 pixel_size1;
[  11.684][d][libmpv_render] [ 22] void main() {
[  11.684][d][libmpv_render] [ 23] vec4 color = vec4(0.0, 0.0, 0.0, 1.0);
[  11.684][d][libmpv_render] [ 24] color.r = 1.003906 * vec4(texture(texture0, texcoord0)).r;
[  11.684][d][libmpv_render] [ 25] color.gb = 1.003906 * vec4(texture(texture1, texcoord1)).rg;
[  11.684][d][libmpv_render] [ 26] color = color.rgbr;
[  11.684][d][libmpv_render] [ 27] color.rgb = mat3(colormatrix) * color.rgb + colormatrix_c;
[  11.684][d][libmpv_render] [ 28] color.a = 1.0;
[  11.684][d][libmpv_render] [ 29] // color mapping
[  11.684][d][libmpv_render] [ 30] // linearize

// HLG inverse EOTF function
[  11.684][d][libmpv_render] [ 31] color.rgb = clamp(color.rgb, 0.0, 1.0);
[  11.684][d][libmpv_render] [ 32] color.rgb = mix(vec3(4.0) * color.rgb * color.rgb,
[  11.684][d][libmpv_render] [ 33]                 exp((color.rgb - vec3(0.559911)) * vec3(1.0/0.178833)) + vec3(0.284669),
[  11.684][d][libmpv_render] [ 34]                 bvec3(lessThan(vec3(0.5), color.rgb)));
[  11.684][d][libmpv_render] [ 35] color.rgb *= vec3(1.0/3.179550);
[  11.684][d][libmpv_render] [ 36] color.rgb *= vec3(1.0/3.774119);
[  11.684][d][libmpv_render] [ 37] color.rgb *= vec3(3.774119);

// HLG reference OOTF function
[  11.684][d][libmpv_render] [ 38] // apply ootf
[  11.684][d][libmpv_render] [ 39] color.rgb *= vec3(1.000749 * pow(dot(src_luma, color.rgb), 0.200000));

[  11.684][d][libmpv_render] [ 40] // HDR tone mapping
[  11.684][d][libmpv_render] [ 41] int sig_idx = 0;
[  11.684][d][libmpv_render] [ 42] if (color[1] > color[sig_idx]) sig_idx = 1;
[  11.684][d][libmpv_render] [ 43] if (color[2] > color[sig_idx]) sig_idx = 2;
[  11.684][d][libmpv_render] [ 44] float sig_max = color[sig_idx];
[  11.684][d][libmpv_render] [ 45] float sig_peak = 4.926108;
[  11.684][d][libmpv_render] [ 46] float sig_avg = 0.250000;
[  11.684][d][libmpv_render] [ 47] vec3 sig = min(color.rgb, sig_peak);
[  11.684][d][libmpv_render] [ 48] float sig_orig = sig[sig_idx];
[  11.684][d][libmpv_render] [ 49] float slope = min(1.000000, 0.250000 / sig_avg);
[  11.684][d][libmpv_render] [ 50] sig *= slope;
[  11.684][d][libmpv_render] [ 51] sig_peak *= slope;

// tone-mapping=linear and tone-mapping-param=peak
[  11.684][d][libmpv_render] [ 52] sig = min(4.926108 / sig_peak, 1.0) * sig;
[  11.684][d][libmpv_render] [ 53] vec3 sig_lin = color.rgb * (sig[sig_idx] / sig_orig);
[  11.684][d][libmpv_render] [ 54] float coeff = max(sig[sig_idx] - 0.180000, 1e-6) /               max(sig[sig_idx], 1.0);
[  11.684][d][libmpv_render] [ 55] coeff = 0.750000 * pow(coeff, 1.500000);
[  11.684][d][libmpv_render] [ 56] color.rgb = mix(sig_lin, 1.000000 * sig, coeff);
[  11.684][d][libmpv_render] [ 57] color.rgb *= vec3(1.000000);
[  11.684][d][libmpv_render] [ 58] out_color = color;
[  11.684][d][libmpv_render] [ 59] }
```

</details>